src/app/user/login/login.component.ts

```diff
@@ -7,22 +10,80 @@ import { Component, OnInit } from '@angular/core';
 })
 
 export class LoginComponent implements OnInit {
-  hide = true;
-  loginError = false;
+  hidePassword = true;
+  errorMessage = "";
+  showLoadingDiv = true;
 
-  testEmail = ["rtoche@uark.edu"]
-  testPassword = ["legomyego"]
+  signInForm: FormGroup;
+  userEmailControl: FormControl;
+  userPasswordControl: FormControl;
 
-  constructor() { }
+  constructor(public auth: AngularFireAuth) { 
+    // Defining user email control
+    this.userEmailControl = new FormControl('', [
+      Validators.required,
+      Validators.email,
 
-  ngOnInit(): void {
+	  ]);
+    // this.userEmailControl.setErrors({customEmailError: false });
+    
+    // Defining password control
+    this.userPasswordControl = new FormControl('', [
+      Validators.required
+		
+	]);
+
+	// Defining sign in form
+	this.signInForm = new FormGroup({
+	  userEmailControl: this.userEmailControl,
+	  userPasswordControl: this.userPasswordControl
+
+	});
+
+	this.showLoadingDiv = false;
+
+  }
+
+  // Keep for future reference: How to create custom validator
+  // https://dzone.com/articles/how-to-create-custom-validators-in-angular
+  emailValidator(control: AbstractControl): {[key: string]: boolean} | null {
+	console.log("EMAIL VALIDATOR")
+	  return null;
   }
 
-  login(userEmail: string, userPassword: string):void {
-    console.log(userEmail);
-    console.log(userPassword);
-    this.loginError = true;
+  ngOnInit(): void {
   }
 
+  signIn(userEmail: string, userPassword: string):void {
+    // Check to make sure form is not invalid before submitting
+    if(!this.signInForm.invalid) {
+      this.showLoadingDiv = true;
+      this.errorMessage = "";
 
+      this.auth.signInWithEmailAndPassword(userEmail, userPassword).then((user) => {
+        // Successfully signed in
+        console.log("Signed in!");
+        
+        this.errorMessage = ""
+        window.location.href = "/"
```

Comment:
didn't catch this earlier, redirect with angular Router as opposed to window.location.href: https://angular.io/api/router/Router